include/mbgl/util/logging.hpp

```diff
@@ -75,13 +75,12 @@ class Log {
     static Log* get() noexcept;
 
     static void record(EventSeverity severity, Event event, const std::string &msg);
-    static void record(EventSeverity severity, Event event, const char* format = "", ...);
-    static void record(EventSeverity severity, Event event, int64_t code, const char* format = "", ...);
+    static void record(EventSeverity severity, Event event, int64_t code, const std::string &msg);
     static void record(EventSeverity severity,
                        Event event,
                        int64_t code,
                        const std::string& msg,
-                       const optional<std::string>& threadName);
+                       const std::optional<std::string>& threadName);
```

Comment:
Build is failing (Node on Windows with `MBGL_EGL = ON`), seems to be affected by these changes, any ideas?

```
C:\builds\maplibre-gl-native\include\mbgl/util/logging.hpp(70): error C2665: 'mbgl::Log::record': none of the 3 overloads could convert all the argument types
C:\builds\maplibre-gl-native\include\mbgl/util/logging.hpp(78): note: could be 'void mbgl::Log::record(mbgl::EventSeverity,mbgl::Event,int64_t,const std::string &)'
C:\builds\maplibre-gl-native\include\mbgl/util/logging.hpp(66): note: while trying to match the argument list '(mbgl::EventSeverity, mbgl::Event, const char [48], _Ty)'
        with
        [
            _Ty=EGLint
        ]
C:\builds\maplibre-gl-native\include\mbgl/util/logging.hpp(61): note: see reference to function template instantiation 'void mbgl::Log::Record<const char(&)[48],_Ty>(mbgl::EventSeverity,mbgl::Event,const char (&)[48],_Ty &&)' being compiled
        with
        [
            _Ty=EGLint
        ]
C:\builds\maplibre-gl-native\platform\windows\src\headless_backend_egl.cpp(34): note: see reference to function template instantiation 'void mbgl::Log::Error<const char(&)[48],EGLint>(mbgl::Event,const char (&)[48],EGLint &&)' being compiled
```

Reply:
We need to update CI so that this can be catched.

There is a logging statement there that uses sprintf. Needs to be rewritten to pass a std::string instead.

Reply:
@etnav 

Could you try this and maybe submit a PR if you get it to build? I am not on Windows.

```
            mbgl::Log::Error(mbgl::Event::OpenGL, "eglBindAPI(EGL_OPENGL_ES_API) returned error " + std::to_string(eglGetError()));
```